coursier.bzl

```diff
@@ -272,6 +272,35 @@ def _pinned_coursier_fetch_impl(repository_ctx):
                 False,  # not executable
             )
 
+def remove_auth_from_url(url):
+    """Returns url without `user:pass@` or `user@`."""
+    if "@" not in url:
+        return url
+    protocol = url[:url.find("://")]
+    url_without_protocol = url[url.find("://") + 3:]
+    url_parts = url_without_protocol.split("/")
+    host = url_parts[0]
+    if "@" not in host:
+        return url
+    host = host[host.find("@") + 1:]
+    new_url = "{}://{}".format(protocol, "/".join([host] + url_parts[1:]))
+    return new_url
+
+def infer_artifact_path_from_primary_and_repos(primary_url, repository_urls):
+    """Returns the artifact path inferred by comparing primary_url with urls in repository_urls.
+
+    Returns:
+        String of the artifact path used by maven to find a particular artifact. Does not have a leading slash (`/`).
+    """
+    repository_urls = [remove_auth_from_url(r.rstrip("/")) for r in repository_urls]
+    primary_url = remove_auth_from_url(primary_url)
```

Comment:
nit: prefer creating new variables instead of reassignment

Reply:
Done.